addons/web_editor/static/src/js/wysiwyg/widgets/image_crop.js

```diff
@@ -97,6 +97,43 @@ export class ImageCrop extends Component {
         }
     }
 
+    /**
+     * Crops image into 1:1 before application of shape if shape has attribute
+     * "data-unstretch=true".
+     *
+     *  @param {boolean} isCropRequired
+     *  @param {boolean} [preview=false]
+     */
+    async cropSquare(isCropRequired, preview = false) {
+        const ratio = isCropRequired ? "1/1" : "0/0";
+        if (this.$cropperImage) {
+            this.$cropperImage.cropper("reset");
+            if (this.aspectRatio !== ratio) {
+                this.aspectRatio = preview ? this.aspectRatio : ratio;
+                this.$cropperImage.cropper("setAspectRatio", this.aspectRatios[ratio].value);
+            }
+            await this._save(isCropRequired);
+        }
+    }
+
+    /**
+     * Undo the preview of shape.
+     */
+    async resetCropSquare() {
+        if (this.media.classList.contains("o_we_image_cropped_preview")) {
+            if (this.$cropperImage) {
+                this.$cropperImage.cropper("reset");
+                this.$cropperImage.cropper(
+                    "setAspectRatio",
+                    this.aspectRatios[this.aspectRatio].value
+                );
+                this.aspectRatio !== "0/0" ? await this._save(true) : await this._save(false);
+            }
+        } else {
+            this._closeCropper();
+        }
+    }
```

Comment:
Code suggestion:
```js
/**
 * Resets the square crop. Needed when undoing the preview of a shape.
 */
async resetCropSquare() {
    if (this.$cropperImage) {
        this.$cropperImage.cropper("reset");
        this.$cropperImage.cropper("setAspectRatio", this.aspectRatios[this.aspectRatio].value);
        await this._save(false, false);
    }
}
```

Note that this function could be removed and replaced by just calling `reset`, if you use the styles instead of the cropper for the previews :+1: